### DISCUSSION TITLE
auto logout function when there is authentication error

### DISCUSSION THREAD

**Discussion by creative2113** [Finalgoal231/finalGoal#104] [Discuss about technical stack] [answered]
Hello, everyone!
I want to add function that auto logout when there is authentication error
If anyone can solve this problem.
please start to work with me.

**Comment by creative2115** [comment, marked as answer]
Hello, @creative2113 
we can use interceptor axiosHandler for solving this problem.
Here is an example code:
```js
import axios from "axios";

const axiosHandler = axios.create({
  baseURL: process.env.REACT_APP_BASE_URL,
  headers: {
    Accept: "application/json",
  },
});

axiosHandler.interceptors.request.use(
  (config) => {
    const token = localStorage.getItem("token");
    if (token) {
      config.headers["Authorization"] = "Bearer " + token;
    }
    return config;
  },
  (error) => {
    Promise.reject(error);
  }
);

//axiosHandler.interceptors.response

export default axiosHandler;
```
in error case, we can add code for logout action in the Error handler field of interceptor.
This code usually is used to handle one kind of error from many apis with same method in case of using axios.
thanks for your reading.

**Reply by creative2113** [reply]
Oh, great!

**Reply by creative2113** [reply]
I will create issue for that discusstion.

**Reply by creative2115** [reply]
thank you.
